src/romi-config.ts

```diff
@@ -0,0 +1,80 @@
+import { DEFAULT_IO_CONFIGURATION, IOPinMode, IPinConfiguration, NUM_CONFIGURABLE_PINS } from "./romi-robot";
+import jsonfile from "jsonfile";
+import ProgramArguments from "./program-arguments";
+
+export interface RomiConfigJson {
+    ioConfig: string[]
+}
+
+export default class RomiConfiguration {
+    private _extIOConfig: IPinConfiguration[] = [];
+
+    constructor(programArgs: ProgramArguments) {
+        // Pre-load the external IO configuration
+        DEFAULT_IO_CONFIGURATION.forEach(val => this._extIOConfig.push(Object.assign({}, val)));
+
+        let isConfigError: boolean = false;
+
+        if (programArgs.config !== undefined) {
+            try {
+                const romiConfig: RomiConfigJson = jsonfile.readFileSync(programArgs.config);
+
+                if (romiConfig && romiConfig.ioConfig && (romiConfig.ioConfig instanceof Array)) {
+                    const portConfigs: string[] = romiConfig.ioConfig;
+                    if (portConfigs.length !== NUM_CONFIGURABLE_PINS) {
+                        isConfigError = true;
+                        throw new Error("[CONFIG] Invalid number of IO port configurations");
+                    }
+
+                    for (let i = 0; i < NUM_CONFIGURABLE_PINS; i++) {
+                        let pinMode: IOPinMode;
+
+                        switch (portConfigs[i].toLowerCase()) {
+                            case "pwm":
+                                pinMode = IOPinMode.PWM;
+                                break;
+                            case "ain":
+                                if (i === 0) {
+                                    isConfigError = true;
+                                    throw new Error("[CONFIG] Invalid mode for pin EXT 0");
```

Comment:
Why can't pin 0 be Analog In?

Reply:
Hardware limitation. It's not connected to a ADC channel